kubert/src/client.rs

```diff
@@ -69,6 +71,40 @@ impl ClientArgs {
         client.try_into().map_err(Into::into)
     }
 
+    /// Initializes a Kubernetes client from a [`tower::Service`].
+    ///
+    /// This will respect the `$KUBECONFIG` environment variable, but otherwise default to
+    /// `~/.kube/config`. The _current-context_ is used unless `context` is set.
+    ///
+    /// This is basically equivalent to using `kube_client::Client::new`, except that it
+    /// supports kubeconfig configuration from the command-line.
+    pub async fn try_from_service<S, B>(self, svc: S) -> Result<Client, ConfigError>
+    where
+        S: Service<Request<Body>, Response = Response<B>> + Send + Clone + 'static,
+        S::Future: Send + 'static,
+        S::Error: Into<BoxError>,
+        B: HttpBody<Data = bytes::Bytes> + Send + 'static,
+        B::Error: Into<BoxError>,
+    {
+        use kube_client::client::{ClientBuilder, ConfigExt};
+
+        let config = match self.load_local_config().await {
+            Ok(client) => client,
+            Err(e) if self.is_customized() => return Err(e),
+            Err(_) => Config::incluster()?,
+        };
+
+        let stack = ServiceBuilder::new()
+            .layer(config.base_uri_layer())
+            // TODO(eliza): add an equivalent gzip config to the one from kube_client?
```

Comment:
Are there barriers to doing this? I'd expect this difference to have potentially substantial performance impacts.

Reply:
The only barrier is adding dependencies. My intention was to prove out that it worked, first, and then add them, but I'm happy to do it in this PR instead.

Reply:
added GZIP behind a feature flag in 8243f20e5e0b954fa66f33db657cb06a3aabdb2d

Reply:
Ah, I see. I assumed it was in their default client features, but it is not. Still it seems like a useful feature, thanks.